lib/web/integrations_awsoidc.go

```diff
@@ -427,6 +429,50 @@ func (h *Handler) awsOIDCConfigureEKSIAM(w http.ResponseWriter, r *http.Request,
 		return nil, trace.BadParameter("invalid role %q", role)
 	}
 
+	var (
+		integrationRoles = make(map[string]struct{})
+		integrations     []types.Integration
+		nextPage         string
+		err              error
+	)
+	// Ensure the passed role exists in an AWS Integration resource. If
+	// it doesn't an attacker could use the created script to give
+	// themselves the permissions that should be granted to an
+	// integration role instead.
+	for {
+		integrations, nextPage, err = h.cfg.AccessPoint.ListIntegrations(r.Context(), 0, nextPage)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		for _, integration := range integrations {
+			if integration.GetSubKind() != types.IntegrationSubKindAWSOIDC {
+				continue
+			}
+			spec := integration.GetAWSOIDCIntegrationSpec()
+			if spec == nil {
+				continue
+			}
+
+			parsedARN, err := arn.Parse(spec.RoleARN)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			// strip 'role/' from the resource name so only the name of
+			// the role is added
+			arnRole := path.Base(parsedARN.Resource)
+			integrationRoles[arnRole] = struct{}{}
+		}
+
+		if nextPage == "" {
+			break
+		}
+	}
+
+	if _, ok := integrationRoles[role]; !ok {
+		return nil, trace.NotFound("role %q not found in existing AWS integration resources", role)
```

Comment:
This could be used by attackers to identify what AWS roles are being used by integrations, but I'm not sure if that helps them much. I originally made the error message identical to if an invalid role was passed above: `invalid role %q` but I thought that would be pretty confusing for users if the cache didn't update fast enough to add a new integration and the passed role couldn't be found in integrations currently in the cache.